plinth/templates/setup.html

```diff
@@ -63,8 +72,17 @@
       <form action="" method="post">
         {% csrf_token %}
 
-        <input type="submit" class="btn btn-md btn-primary"
-               value="{% trans "Update" %}" />
+        {% if dpkg_is_active %}
+          <div class="alert alert-warning" role="alert">
+            Another installation or upgrade is already running.
+            Please wait for a few moments and refresh this page.
+          </div>
```

Comment:
Using an alert is better than messages as you mentioned in the original bug. Message are stored in sessions and are showing next page load which ever that is.